### PR TITLE
592-new-exercises

### PR DESCRIPTION
**Pull request by nicktu12**
## Description of changes made

Creates 3 FP test suites for practice

    Adds video games, dj and favorite foods tests

## Motivation for any changes

Creates tests that involve creating multiple functions, while practicing use of different parameter types, for loops. 

## Questions that you have for a reviewer

Are these tests too easy?
Might need a final PR on this repo to update order of tests, as well as standardize across suites. 

Before merge

[x] add skips
[x] remove solutions

Closes #592 (not linked bc repo not attached to project board)

### CODE REVIEW

File: dj/dj-test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        var reggaeSong = createSong('Rudolph\'s Reggae Jam', 'The Reindeer Rhythms')
+
+        var holidayPlaylist = makePlaylist('Holiday Songs', [rockinSong, shuffleSong, reggaeSong]); 
+
+        var updatedHolidayPlaylist = playAllSongs(holidayPlaylist) 
+
+        assert.equal(updatedHolidayPlaylist.songs[0].hasBeenPlayed, true)
+        assert.equal(updatedHolidayPlaylist.songs[1].hasBeenPlayed, true)
+        assert.equal(updatedHolidayPlaylist.songs[2].hasBeenPlayed, true)
+     })
```
Comment (kaylagordon) [inline dj/dj-test.js:70]: love this one!

File: favorite-foods/favorite-foods-test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+   })
+
+   // spicy 🌶️
+   it.skip('should be able to make a list of all ingredients needed for multiple dishes', function () {
+      var pizza = createFavoriteFood({ dish: 'Pizza', ingredients: ['Tomato Sauce', 'Cheese', 'Pepperoni', 'Mushrooms'], isSpicy: false });
+      var smoothie = createFavoriteFood({ dish: 'Smoothie', ingredients: ['Banana', 'Strawberry', 'Blueberry', 'Milk', 'Honey'], isSpicy: false })
+
+      var list = createShoppingList([pizza, smoothie])
+
+      assert.deepEqual(list, ['Tomato Sauce', 'Cheese', 'Pepperoni', 'Mushrooms', 'Banana', 'Strawberry', 'Blueberry', 'Milk', 'Honey'])
```
Comment (kaylagordon) [inline favorite-foods/favorite-foods-test.js:90]: oo spicy! I like!!

File: video-games/video-games-test.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      assert.equal(level.coins, 0);
+      assert.equal(level.lives, 3);
+
+      var updatedLevelPlayerDefeated = defeatPlayer(level);
+      assert.equal(updatedLevelPlayerDefeated.lives, 2);
+      var updatedLevelPlayerAnnihilated  = defeatPlayer(updatedLevelPlayerDefeated);
+      assert.equal(updatedLevelPlayerAnnihilated.lives, 1);
+      var updatedLevelPlayerDestroyed = defeatPlayer(updatedLevelPlayerAnnihilated);
+      assert.equal(updatedLevelPlayerDestroyed, 'GAME OVER');
+   });
```
Comment (kaylagordon) [inline video-games/video-games-test.js:104]: lol @ these variable names

File: video-games/video-games-test.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      var player1 = createPlayer('The Timeless Troubadour', 500, ['piano sonata', 'violin concerto', 'fllamenco guitar']);
+      var level = createLevel('Infernal Inferno of Flames', [player1]);
+
+      assert.equal(level.name, 'Infernal Inferno of Flames');
+      assert.deepEqual(level.players, [player1]);
+      assert.equal(level.coins, 0);
+      assert.equal(level.lives, 3);
+
+      var updatedLevel = findCoins(level, 5);
+      assert.equal(updatedLevel.coins, 5);
```
Comment (kaylagordon) [inline video-games/video-games-test.js:73]: Might be cool to add a piece where if they get 100 coins, they add a life?

**Review by kaylagordon — COMMENTED** [review]
These exercises look great, Nick!

File: video-games/video-games-test.js (hunk 3 above)
Comment (nicktu12) [reply video-games/video-games-test.js:104]: Trying to enforce some FP! I agree these names are a bit contrived

File: video-games/video-games-test.js (hunk 4 above)
Comment (nicktu12) [reply video-games/video-games-test.js:73]: Love this idea - I will make a quick update :)

**Comment by nicktu12** [comment]
@kaylagordon added one more test for adding lives - thanks for the review!

**Pull request merged**